dell/precision/5560/default.nix

```
@@ -0,0 +1,21 @@
+{ lib, ... }:
+{
+  imports = [
+    ../../../common/pc/laptop
+    ../../../common/pc/laptop/ssd
+    ../../../common/cpu/intel
+    ../../../common/gpu/intel/tiger-lake
+    ../../../common/gpu/nvidia/turing
+  ];
+
+  hardware.enableRedistributableFirmware = lib.mkDefault true;
+
+  boot = {
+    kernelParams = [ "i915.modeset=1" ];
```

@Mic92 what about this? I think it should at least be wrapped in a
`lib.mkIf (config.hardware.intelgpu.driver == "i915")`
but at that point, maybe it should be in `common` ?

---

You are probably right.

```suggestion
    kernelParams = lib.mkIf (config.hardware.intelgpu.driver == "i915") [ "i915.modeset=1" ];
```

Also is this laptop not new enough to use `xe` anyway? Also isn't modset=1 the default since along time also?

```
lsmod | grep 'xe'
xe                   2875392  0
drm_gpuvm              45056  1 xe
drm_exec               12288  2 drm_gpuvm,xe
gpu_sched              65536  1 xe
drm_suballoc_helper    12288  1 xe
drm_ttm_helper         12288  1 xe
drm_buddy              20480  2 xe,i915
ttm                   102400  3 drm_ttm_helper,xe,i915
drm_display_helper    258048  2 xe,i915
cec                    73728  3 drm_display_helper,xe,i915
i2c_algo_bit           20480  2 xe,i915
video                  81920  2 xe,i915
backlight              28672  3 video,xe,i915

```

---

it is, I left a note in the readme about using xe instead of i915
I simply don't know if we should default to that or not, since it's pretty new and might be less stable than i915

---

also, I'm not sure about `i915.modeset`, I can see this locally:
```
$ modinfo i915 | grep modeset:
parm:           modeset:Use kernel modesetting [KMS] (0=disable, 1=on, -1=force vga console preference [default]) (int)
```

---

are you on a precision5560 right now? I had to use `force_probe` to swap to xe

---

`common/gpu/intel/tiger-lake/default.nix` does seem to default to xe when the kernel has support for it

---

No. I am on 12th gen intel framework laptop. However you will get the xe driver by default when you update to linux 6.8, I believe:

https://github.com/NixOS/nixos-hardware/blob/master/common/gpu/intel/tiger-lake/default.nix

---

So maybe just adding the conditional is enough. Please also add the reason, why you did add this flag.

---

I just checked, on 6.10.11 I get this:
```
[    0.829046] xe 0000:00:02.0: Your graphics device 9a60 is not officially supported
               by xe driver in this kernel version. To force Xe probe,
               use xe.force_probe='9a60' and i915.force_probe='!9a60'
               module parameters or CONFIG_DRM_XE_FORCE_PROBE='9a60' and
               CONFIG_DRM_I915_FORCE_PROBE='!9a60' configuration options.
```

---

that being said, I've been using xe since I got this laptop and everything works fine
it _is_ the earliest supported igpu by Xe
should we add the force_probe config here? or in gpu/intel/tigerlake ?

---

Mhm. Maybe limit this to your laptop for now, to reduce the blast radius.

---

fair enough, and thanks for reviewing btw!
but now I'm not sure the conditional in common/gpu/intel/tiger-lake/default.nix makes sense
could there be other tigerlake igpu device IDs that _are_ whitelisted for xe? if not, maybe we should remove xe from that config

---

I looked through the kernel sources for xe, found this struct:
```
static const struct xe_device_desc tgl_desc = {
        .graphics = &graphics_xelp,
        .media = &media_xem,
        PLATFORM(TIGERLAKE),
        .has_display = true,
        .has_llc = true,
        .require_force_probe = true,
};
```
so tigerlake has `require_force_probe = true`

---

Seems like it, only LUNARLAKE and BATTLEMAGE will actually use the driver without force probe: 

```
static const struct xe_device_desc lnl_desc = {
	PLATFORM(LUNARLAKE),
	.has_display = true,
};

static const struct xe_device_desc bmg_desc = {
	DGFX_FEATURES,
	PLATFORM(BATTLEMAGE),
	.has_display = true,
	.has_heci_cscfi = 1,
};
```

So we should probably adapt it for nixos-hardware as well.

---

So this was introduced by @fidgetingbits in https://github.com/NixOS/nixos-hardware/pull/944

---

https://github.com/NixOS/nixos-hardware/pull/1149